camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/catalog/CatalogOsgiLibraryTest.java

```diff
@@ -0,0 +1,286 @@
+import org.apache.brooklyn.api.mgmt.ManagementContext;
+import org.apache.brooklyn.camp.brooklyn.AbstractYamlTest;
+import org.apache.brooklyn.core.config.external.AbstractExternalConfigSupplier;
+import org.apache.brooklyn.core.mgmt.internal.ExternalConfigSupplierRegistry;
+import org.apache.brooklyn.core.mgmt.internal.ManagementContextInternal;
+import org.apache.brooklyn.core.mgmt.osgi.OsgiVersionMoreEntityTest;
+import org.apache.brooklyn.entity.stock.BasicApplication;
+import org.apache.brooklyn.test.Asserts;
+import org.apache.brooklyn.test.support.TestResourceUnavailableException;
+import org.apache.brooklyn.util.core.http.BetterMockWebServer;
+import org.apache.brooklyn.util.stream.Streams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.UrlEscapers;
+import com.google.mockwebserver.Dispatcher;
+import com.google.mockwebserver.MockResponse;
+import com.google.mockwebserver.RecordedRequest;
+
+public class CatalogOsgiLibraryTest extends AbstractYamlTest {
+    
+    @SuppressWarnings("unused")
+    private static final Logger log = LoggerFactory.getLogger(CatalogOsgiLibraryTest.class);
+
+    private BetterMockWebServer webServer;
+
+    private String jarName = "brooklyn-test-osgi-entities.jar";
+    private String malformedJarName = "thisIsNotAJar.jar";
+    private String classpathUrl = "classpath:/brooklyn/osgi/" + jarName;
+    private URL jarUrl;
+    private URL malformedJarUrl;
+    
+    @BeforeMethod(alwaysRun = true)
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        
+        // Load the bytes of the jar
+        TestResourceUnavailableException.throwIfResourceUnavailable(getClass(), "/brooklyn/osgi/" + jarName);
+        InputStream resource = getClass().getResourceAsStream("/brooklyn/osgi/" + jarName);
+        final byte[] jarBytes = Streams.readFullyAndClose(resource);
+        
+        // Start a mock web-server that will return the jar
+        webServer = BetterMockWebServer.newInstanceLocalhost();
+        webServer.setDispatcher(new Dispatcher() {
+            @Override
+            public MockResponse dispatch(RecordedRequest request) throws InterruptedException {
+                if (request.getPath().contains(jarName)) {
+                    return new MockResponse().setResponseCode(200).setBody(jarBytes);
+                } else if (request.getPath().contains(malformedJarName)) {
+                    return new MockResponse().setResponseCode(200).setBody("simulating-malformed-jar");
+                } else {
+                    return new MockResponse().setResponseCode(404).setBody(request.getPath() + " not found");
+                }
+            }});
```

Comment:
FYI, there's [TestHttpServer](https://github.com/apache/brooklyn-server/blob/master/utils/common/src/main/java/org/apache/brooklyn/test/http/TestHttpServer.java) which is [a bit easier to set up](https://github.com/apache/brooklyn-server/blob/master/core/src/test/java/org/apache/brooklyn/util/core/ResourceUtilsHttpTest.java#L53-L68).